src/langpacks.py

```diff
@@ -0,0 +1,332 @@
+                    shutil.rmtree(builddir)
+                    logger.debug(f"Removed the existing cache directory: {builddir}")
+                except OSError as e:
+                    logger.error(f"Failed to remove cache directory {builddir}: {e}")
+
+    def build_langpacks(self, base, release):
+        """Build the langpacks."""
+        lp = Launchpad.login_anonymously("langpacks", "production")
+        ubuntu = lp.distributions["ubuntu"]
+
+        # check that the series used is valid
+        active_series = []
+        for s in ubuntu.series:
+            if s.active:
+                active_series.append(s.name)
+
+        release = release.lower()
+        devel_series = ubuntu.getDevelopmentSeries()[0].name
+        if release == "devel":
+            release = devel_series
+
+        if release not in active_series:
+            logger.debug(f"Release {release} isn't an active Ubuntu series")
+            return
+
+        releasedir = HOME / release
+        if not os.path.exists(releasedir):
+            # Create target directory
+            try:
+                run(
+                    [
+                        "sudo",
+                        "-u",
+                        "ubuntu",
+                        "mkdir",
+                        HOME / release,
+                    ],
+                    check=True,
+                    stdout=PIPE,
+                    stderr=STDOUT,
+                    text=True,
+                )
+                logger.debug(f"Directory {HOME / release} created.")
+            except CalledProcessError as e:
+                logger.debug(f"Creating directory failed: {e.stdout}")
+                raise
+
+        if base:
+            download_url = (
+                f"https://translations.launchpad.net/ubuntu/{release}/+latest-full-language-pack"
+            )
+            tarball = REPO_LOCATION / f"ubuntu-{release}-translations.tar.gz"
+            import_options = ["-v", "--treshold=10"]
+
+            # Clean existing cache directories before starting a base build
+            self._clean_builddir(releasedir)
+
+        else:
+            download_url = (
+                f"https://translations.launchpad.net/ubuntu/{release}/+latest-delta-language-pack"
+            )
+            tarball = REPO_LOCATION / f"ubuntu-{release}-translations-update.tar.gz"
+            import_options = ["-v", "--update", "--treshold=10"]
+
+        # Download the current translations tarball from launchpad
+        try:
+            run(
+                [
+                    "sudo",
+                    "-u",
+                    "ubuntu",
+                    "wget",
+                    "--no-check-certificate",
+                    "-q",
+                    "-O",
+                    tarball,
+                    download_url,
+                ],
+                check=True,
+                stdout=PIPE,
+                stderr=STDOUT,
+                text=True,
+            )
```

Comment:
Why call out to `wget` here and not use something like `requests`?

Reply:
Because the python code is running as root and the service/data is owned by the "ubuntu" user. I don't know if there is a better way to handle that in the charm...
(it would be easier if the service was running into the root account the container but it is building deb packages and the Debian packaging tools strongly encourage to do builds as non privileged users)

Reply:
Resolved, not trying to run as the ubuntu user allowed to use requests directly